mojo/core/broker_castanets.cc

```diff
@@ -323,10 +331,49 @@ void BrokerCastanets::SyncSharedBufferImpl(const base::UnguessableToken& guid,
                                            size_t offset,
                                            size_t sync_size,
                                            size_t mapped_size,
+                                           BrokerCompressionData compression_data,
                                            bool write_lock) {
   CHECK_GE(mapped_size, offset + sync_size);
   BufferSyncData* buffer_sync = nullptr;
   void* extra_data = nullptr;
+  uint8_t* start_ptr = static_cast <uint8_t*>(memory + offset);
+  uint8_t* compressed_data = nullptr;
+  int original_sync_size = sync_size;
+  std::string compressed;
+  BrokerCompressionMode compression_mode = compression_data.mode();
+  if (sync_size <= COMPRESSION_THRESHOLD_BYTES)
+    compression_mode = NONE;
+
```

Comment:
I didn’t observe such case in webp. Only if the original size is too small this case is observed in zlib.  As we use webp only for tiles, it may have sync size of at least 16384 bytes (for 64 size tile).
The quality factor is considered as 100. Should that be ok ?

Reply:
1. Can we apply 'compression_mode = NONE' only for the Zlib as this also applies to WebP?
2. For WebP, the current code is WebPEncodeRGBA with 100. Why don't we go for WebPEncodeLosslessRGBA? quality factor 100 is still observed poor comparing to lossless.

Reply:
Yes right, I have updated it to use lossless api.